osf/models/preprint_provider.py

```diff
@@ -17,6 +20,36 @@ class PreprintProvider(ObjectIDMixin, BaseModel):
     banner_name = models.CharField(null=True, blank=True, max_length=128)  # max length on prod: 19
     external_url = models.URLField(null=True, blank=True, max_length=200)  # max length on prod: 25
 
+    email_contact = models.CharField(null=True, blank=True, max_length=128)  # max length on prod: 23
+    email_support = models.CharField(null=True, blank=True, max_length=128)  # max length on prod: 23
+    advisory_board = models.TextField(null=True, blank=True, max_length=2048)  # max length on prod: 1689
+    social_twitter = models.CharField(null=True, blank=True, max_length=128)  # max length on prod: 8
+    social_facebook = models.CharField(null=True, blank=True, max_length=128)  # max length on prod: 8
+    social_instagram = models.CharField(null=True, blank=True, max_length=128)  # max length on prod: 8
+    subjects_acceptable = DateTimeAwareJSONField(blank=True, default=list)
+    licenses_acceptable = models.ManyToManyField('NodeLicense', related_name='licenses')
+
+    @property
+    def top_level_subjects(self):
+        if len(self.subjects_acceptable) == 0:
+            return Subject.objects.filter(parents=None)
+        tops = set([sub[0][0] for sub in self.subjects_acceptable])
+        return [Subject.load(sub) for sub in tops]
+
+    @property
+    def all_subjects(self):
```

Comment:
This could probably be optimized (later) into one sql query.

Right now it seems like some black magic 😼. Maybe add a few comments about what it's doing, while it's still in your head, so whoever optimizes it doesn't have to step through it trying to figure out what's going on?